pythonforandroid/recipes/beautifulsoup4/__init__.py

```diff
@@ -0,0 +1,25 @@
+ 
+from pythonforandroid.toolchain import (PythonRecipe, shprint,
+                                        current_directory, info, Recipe)
+from pythonforandroid.patching import will_build, check_any
+import sh
+from os.path import join
+
+class BSoup4Recipe(PythonRecipe):
```

Comment:
The fact that you need to set the compiler stuff seems to suggest that this is not a pure python recipe, so maybe you need to use `CythonRecipe` or `CompiledComponentsPythonRecipe` instead.